### PR TITLE
Single test suites source per execution

### PR DESCRIPTION
**Pull request by nulls**
Agreed with @akuleshov7 to avoid generating `save.toml` on orchestrator level and run test suites as is. It can be done only for a single test suites source

### What's done:
- supported case when agent sends empty result to backend
- added new endpoint to download test suites source by execution
- modified TestBatch: now it's only list of tests
- removed synthetic toml, will be handled properly in https://github.com/saveourtool/save-cli/issues/430

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestExecutionController.kt [hunk 1]
```diff
@@ -226,7 +226,9 @@ class TestExecutionController(private val testExecutionService: TestExecutionSer
      */
     @PostMapping(value = ["/internal/saveTestResult"])
     fun saveTestResult(@RequestBody testExecutionsDto: List<TestExecutionDto>) = try {
-        if (testExecutionService.saveTestResult(testExecutionsDto).isEmpty()) {
+        if (testExecutionsDto.isEmpty()) {
```
Comment (kgevorkyan) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestExecutionController.kt:229]: But in which case they will sent empty results?

**Review by kgevorkyan — APPROVED** [review]

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestExecutionController.kt (hunk 1 above)
Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestExecutionController.kt:229]: originally, I found it after issue in logic, but we have ability to exclude tests on test suite level. so we can select test suite with disabled tests and they can be run on a single agent (theoretically 😄 )

File: save-orchestrator/src/main/kotlin/com/saveourtool/save/orchestrator/service/DockerService.kt [hunk 2]
```diff
@@ -222,13 +220,8 @@ class DockerService(
...
     )
     private fun prepareImageAndVolumeForExecution(resourcesForExecution: Path, execution: Execution): RunConfiguration<PersistentVolumeId> {
-        // create stub toml config in aim to execute all test suites directories from `testSuitesDir`
-        resourcesForExecution.resolve(TEST_SUITES_DIR_NAME)
-            .resolve("save.toml")
-            .apply { createFile() }
-            .writeText(createSyntheticTomlConfig(execution.execCmd, execution.batchSizeForAnalyzer))
         // collect test suite names, which were selected by user
-        val saveCliExecFlags = " $TEST_SUITES_DIR_NAME --include-suites \"${execution.getTestSuiteNames()}\""
+        val saveCliExecFlags = " --include-suites \"${execution.getTestSuiteNames().joinToString(DATABASE_DELIMITER)}\" $TEST_SUITES_DIR_NAME"
```
Comment (nulls) [inline save-orchestrator/src/main/kotlin/com/saveourtool/save/orchestrator/service/DockerService.kt:224]: the main problem -- `joinToString` wasn't called

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestSuitesSourceController.kt [hunk 3]
```diff
@@ -328,6 +314,41 @@ class TestSuitesSourceController(
...
+    @ApiResponse(responseCode = "409", description = "IDs of test suites were not set on requested execution or weren't found by ID.")
+    @ApiResponse(responseCode = "404", description = "Test suites source with such name in organization name was not found.")
+    @ApiResponse(responseCode = "409", description = "Organization was not found by provided name.")
+    fun downloadByExecutionId(
+        @RequestParam executionId: Long
+    ): Mono<ByteBufferFluxResponse> = blockingToMono {
+        val execution = executionService.findExecution(executionId)
+            .orNotFound { "Execution (id=$executionId) not found" }
+        val testSuiteId = execution.parseAndGetTestSuiteIds()?.firstOrNull()
+            .orConflict { "Execution (id=$executionId) doesn't contain testSuiteIds" }
```
Comment (kgevorkyan) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestSuitesSourceController.kt:343]: Why `conflict` and not `not found`?

**Review by kgevorkyan — APPROVED** [review]

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestSuitesSourceController.kt [hunk 4]
```diff
@@ -328,6 +314,41 @@ class TestSuitesSourceController(
         }
         .map { it.toDto() }
 
+    /**
+     * @param executionId
+     * @return selected [TestSuitesSourceDto] with versions for [com.saveourtool.save.entities.Execution] found by provided values
+     */
+    @PostMapping("/download-snapshot-by-execution-id", produces = [MediaType.APPLICATION_OCTET_STREAM_VALUE])
```
Comment (kgevorkyan) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestSuitesSourceController.kt:321]: Btw, we decide to use swagger annotations, but it clogs up so much, as for me 👎

Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestSuitesSourceController.kt:321]: +1

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestSuitesSourceController.kt (hunk 3 above)
Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestSuitesSourceController.kt:343]: I think it's a conflict situation, when we request download snapshot from execution which doesn't contain testSuiteIds. It means that we request this endpoint before execution is properly populated

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestSuitesSourceController.kt (hunk 4 above)
Comment (kgevorkyan) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestSuitesSourceController.kt:321]: lines 317-336 - blablabla
lines 337-350 - code

**Pull request merged**